### ISSUE TITLE
Forms failing to announce errors on submit (Phase 1)

### ISSUE THREAD

**Issue by amyleadem** [department-of-veterans-affairs/vets-design-system-documentation#5175]
## Description
Fix the bug in the forms library that prevents errors from being announced on submit. For phase 1, implement fixes for the following components: 

- va-checkbox
- va-checkbox-group
- va-combo-box
- va-radio
- va-radio-option
- va-select
- va-statement-of-truth
- va-text-input
- va-textarea

Other error announcement bugs will be addressed in future issues.

## Tasks
- [ ] Create a fix that reliably announces errors and manages focus for the components listed above
- [ ] Update e2e tests
- [ ] Perform rigorous testing in the following combinations:
    - [ ] Mac OS with VoiceOver: Firefox, Safari, Chrome
    - [ ] Windows with JAWS and NVDA: Chrome and Firefox
    - [ ] iOS with VoiceOver Gestures: Safari
    - [ ] Android with Talkback: Chrome
- [ ] When applicable, inform downstream teams that they can close their related issues

## Acceptance Criteria
Interaction requirements for errors on submit:
- [ ] When the user attempts to advance to the next page by pressing the continue or submit button, AND there is an error present on the page:  
    - [ ] The user should remain on the page with the error. They should not advance to the next page.  
    - [ ] The user’s focus should be programmatically moved to the first erroring element on the page. 
        - [ ] All users should be placed directly inside the erroring input, or in rare cases, be able to easily navigate to the erroring input by pressing the tab key once.   
        - [ ] Sighted users should see an error message that has distinct styling that conveys an error state. The visual error message should be visually associated with the erroring element.  
            - [ ] Users who use page zoom or text resizing should also clearly understand this visual association 
        - [ ] Screen reader users should hear that there is an error on the page
    - [ ] The error message should give a clear, succinct description of the error and how to fix it  
- [ ] This behavior should happen on every submit attempt
- [ ] The error message should be removed for all users when the error is resolved
